daemon/daemon/daemon.go

```diff
@@ -523,22 +526,73 @@ func (d *Daemon) staleMapWalker(path string) error {
 func changedOption(key string, value bool, data interface{}) {
 }
 
-func (d *Daemon) Update(opts option.OptionMap) error {
+type putConfig struct {
+	daemon *Daemon
+}
+
+func NewPutConfigHandler(d *Daemon) PutConfigHandler {
+	return &putConfig{daemon: d}
+}
+
+func (h *putConfig) Handle(params PutConfigParams) middleware.Responder {
+	d := h.daemon
+	opts := params.Configuration
+
 	d.conf.OptsMU.Lock()
 	defer d.conf.OptsMU.Unlock()
 
 	if err := d.conf.Opts.Validate(opts); err != nil {
-		return err
+		return NewApiError(400, err)
```

Comment:
I'm not quite happy with the error handling yet. Ideally all errors are defined in the API spec so users can act upon particular errors with well known error codes. Doing so well generate a type for each and the client side will convert it back to a type with a corresponding name, e.g. `PutEndpointAlreadyExists`